roachtest: allow more errors for REGION survivability DRT

otan opened the pull request:
Region survivability seems to take a bit more errors - presumably, cross
region survivability takes a bit longer to recover from. As such,
increase the fudge factor for errors.

Resolves #74742

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80526)
<!-- Reviewable:end -->

otan committed:
roachtest: allow more errors for REGION survivability DRT

Region survivability seems to take a bit more errors - presumably, cross
region survivability takes a bit longer to recover from. As such,
increase the fudge factor for errors.

Release note: None

otan commented:
bors r=rafiss

irfansharif commented:
blathers backport 22.1
